### PR TITLE
✨ os: add nfs resource for exports and client mounts

### PR DESCRIPTION
**Pull request by tas50**
## Summary
- Adds an `nfs` resource to the os provider with two query surfaces: `exports` (server-side, parsed from `/etc/exports`) and `mounts` (client-side, filtered from the system mount table). Supports Linux, FreeBSD, macOS, and AIX.
- Each export is emitted as one row per `(path, client)` pair with derived `readOnly` and `noRootSquash` flags, so audits like `nfs.exports.where(noRootSquash)` work directly across all four platforms despite the underlying `/etc/exports` syntaxes being meaningfully different.
- Each mount surfaces parsed NFS-specific fields: protocol `version`, `security` flavor, `hardMount`, `readOnly`, plus split `server` / `remotePath` (IPv6-aware) and the full raw option list.

## Why
The existing `mount` resource already shows NFS mounts in the generic mount table, but you'd have to string-match raw options to write any meaningful audit. There was no view at all of the server side. CIS- and DISA-style NFS findings (e.g. _"no export grants `no_root_squash`"_, _"NFS mounts use sec=krb5 or stronger"_, _"NFS mounts use hard semantics"_) need parsed primitives, not raw option text — and they need to mean the same thing on Linux (`no_root_squash`), BSD/macOS (`-maproot=root`), and AIX (`-root=<hostlist>`).

## Resource shape
```
nfs {
  exports() []nfs.export
  mounts()  []nfs.mount
}

private nfs.export {
  path, client, options []string, readOnly, noRootSquash
}

private nfs.mount {
  device, mountpoint, server, remotePath,
  version, security, hardMount, readOnly, options []string
}
```

Synthetic `__id`s are hidden (`<path>@<client>` for exports, `mountpoint` for mounts) per the project convention.

## Implementation notes
- Parser package lives at `providers/os/resources/nfs/` with one file per syntax family (`linux_exports.go`, `bsd_exports.go`, `aix_exports.go`) plus a shared dispatcher and mount-side helpers. Filenames intentionally avoid the trailing `_<GOOS>.go` form so Go doesn't apply implicit OS build constraints — this is parsing logic, not platform code.
- BSD parser handles both FreeBSD's `-flag=value` and macOS's `-flag value` styles, recognizes the NFSv4 `V4:` root line, and accepts multiple paths per line (a real BSD feature).
- AIX parser derives `readOnly` and `noRootSquash` per-host by intersecting each host with the `ro=`/`rw=`/`root=`/`access=` hostlists, since on AIX those attributes are properties of the (export, host) pair rather than of the export as a whole.
- Mount resource reuses the existing `mount.OperatingSystemMountManager`, filters by `IsNFSFsType` (handles `nfs`, `nfs3`, `nfs4`, `nfsv4`, etc.), and falls back to fstype-derived version when the `vers=` option is absent.

## Test plan
- [x] `go test ./providers/os/resources/nfs/` — 30+ subtests across exports parsing (all 4 platforms, edge cases: comments, blank lines, multi-client/multi-path lines, IPv6 device strings, version fallback, soft/hard semantics)
- [x] `go test ./providers/os/resources/...` — full os provider suite, no regressions
- [x] `make providers/build/os` — clean build
- [x] `gofmt -l` and `go vet` — clean
- [ ] Interactive smoke test on a Linux host with `/etc/exports` populated: `cnspec shell local -c "nfs.exports { path client readOnly noRootSquash }"`
- [ ] Interactive smoke test against a macOS or FreeBSD host with at least one export and at least one NFS mount

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: providers/os/resources/nfs/bsd_exports.go [hunk 1]
```diff
@@ -0,0 +1,127 @@
...
+			pathPhase = false
+			hosts = append(hosts, tok)
+		}
+	}
+	return paths, options, hosts
+}
+
+func bsdReadOnly(options []string) bool {
+	for _, o := range options {
+		if o == "-ro" || o == "-o" {
```
[the remark this answers is not in the document]
Comment (tas50) [reply providers/os/resources/nfs/bsd_exports.go]: Fixed in b949070 — removed the `-o` clause; only `-ro` marks BSD exports read-only.

File: providers/os/resources/nfs.go [hunk 2]
```diff
@@ -0,0 +1,139 @@
...
+	}
+
+	out := make([]any, 0, len(entries))
+	for _, e := range entries {
+		options := make([]any, len(e.Options))
+		for i, o := range e.Options {
+			options[i] = o
+		}
+		res, err := CreateResource(n.MqlRuntime, "nfs.export", map[string]*llx.RawData{
+			"__id":         llx.StringData(fmt.Sprintf("%s@%s", e.Path, e.Client)),
```
[the remark this answers is not in the document]
Comment (tas50) [reply providers/os/resources/nfs.go]: Fixed in b949070 — switched the separator from `@` to `\x00`, matching the separator the runtime already uses for cache keys. `\x00` can't appear in either field. On the duplicate-line concern: a path/client pair appearing in multiple `/etc/exports` lines with different options is undefined per exports(5) — the kernel takes one — so we don't need to model the duplicate-row case.

File: providers/os/resources/nfs.go [hunk 3]
```diff
@@ -0,0 +1,139 @@
...
+func (n *mqlNfs) mounts() ([]any, error) {
+	conn, ok := n.MqlRuntime.Connection.(shared.Connection)
+	if !ok {
+		return nil, errors.New("wrong connection type")
+	}
+
+	mm, err := mount.ResolveManager(conn)
+	if mm == nil || err != nil {
+		return nil, fmt.Errorf("could not detect suitable mount manager for platform")
+	}
```
[the remark this answers is not in the document]
Comment (tas50) [reply providers/os/resources/nfs.go:86]: Fixed in b949070 — split the nil-manager and error branches and used `%w` to preserve the original cause for `mount.ResolveManager` and `mm.List`.

File: providers/os/resources/nfs/linux_exports.go [hunk 4]
```diff
@@ -0,0 +1,120 @@
...
+
+import (
+	"io"
+	"strings"
+)
+
+// parseLinuxExports parses /etc/exports in the syntax used by the
+// Linux NFS server (nfs-utils). Each line is
+//
+//	path  client1(opt,opt,...) client2(opt,opt,...)
```
[the remark this answers is not in the document]
Comment (tas50) [reply providers/os/resources/nfs/linux_exports.go:14]: Fixed in b949070 — `scanExportLines` now joins lines ending in `\` before parsing. Added a test in `exports_test.go` covering a continuation across three physical lines. (Plus a netgroup-client test since that came up in the other thread.)

File: providers/os/resources/nfs.go [hunk 5]
```diff
@@ -0,0 +1,218 @@
...
+	if err != nil {
+		return nil, err
+	}
+
+	out := make([]any, 0, len(entries))
+	for _, e := range entries {
+		options := make([]any, len(e.Options))
+		for i, o := range e.Options {
+			options[i] = o
+		}
```
[the remark this answers is not in the document]
Comment (tas50) [reply providers/os/resources/nfs.go:57]: This was discussed in the earlier thread on the `__id` field (https://github.com/mondoohq/mql/pull/7956#discussion_r3301699901): two `/etc/exports` lines for the same `(path, client)` pair with different options is undefined per exports(5) — the kernel takes one — so the resource intentionally collapses the duplicate to a single row rather than synthesizing a disambiguating id. Leaving the behavior as-is.

The version-number concern is fixed in the latest force-push (now `13.19.1`).
